src/app/ReadHandler.cpp

```diff
@@ -226,6 +225,12 @@ CHIP_ERROR ReadHandler::SendReportData(System::PacketBufferHandle && aPayload, b
     }
 
     VerifyOrReturnLogError(mpExchangeCtx != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    // If mIsChunkedReport is not set, then we are generating our first report chunk, record the current tick for checking dirty
+    // read handlers.
+    if (!mIsChunkedReport)
```

Comment:
See my comment later on about how this isn't a very easy to understand way of formulating "is this the first report in a chunked series?".